projectroles/utils.py

```diff
@@ -95,3 +109,197 @@ def get_app_names():
             else:
                 ret.append(s[0])
     return sorted(ret)
+
+
+class SidebarContent:
+    """Class for generating sidebar content for the site"""
+
+    def _is_active(self, request, link_names=None):
+        """Check if current URL is active in the sidebar."""
+        # HACK: Avoid circular import
+        from projectroles.urls import urlpatterns
+
+        if request.resolver_match.app_name != 'projectroles':
+            return False
+        url_name = request.resolver_match.url_name
+        if url_name in [u.name for u in urlpatterns]:
+            if link_names:
+                if not isinstance(link_names, list):
+                    link_names = [link_names]
+                if url_name not in link_names:
+                    return False
+            return True
+        return False
+
+    def _is_active_plugin(self, app_plugin, app_name, url_name):
+        """
+        Check if current URL is active in the sidebar for a specific app plugin.
+        """
+        if app_plugin.name.startswith(app_name) and url_name in [
+            u.name for u in getattr(app_plugin, 'urls', [])
+        ]:
+            return True
+        # HACK for remote site views, see issue #1336
+        if (
+            app_name == 'projectroles'
+            and app_plugin.name == 'remotesites'
```

Comment:
Note that the `get_app_link_state()` template tag is used for both sidebar and user dropdown links. The remote sites part here only applies to the user dropdown.

We shouldn't have needless repetition either. The optimal thing would be to have a single helper class for both the sidebar and the user dropdown retrieval, where both API methods would use this `_is_active_plugin()` helper. This is something I overlooked in my original specs..